### PR TITLE
Initialize a URL from a static string

### PR DESCRIPTION
**Pull request by MaxHasADHD**

### CODE REVIEW

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          5 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/Array+UtilitiesTests.swift#L100">Array+UtilitiesTests.swift#L100</a> - <strong><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L100">SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L100</a></strong>: ‘subscript(ip_safe:)’ is deprecated: Use <code>ip_safely</code> instead.  <br /><code>
        XCTAssert(testArray[ip_safe: 3] == 4)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/Array+UtilitiesTests.swift#L101">Array+UtilitiesTests.swift#L101</a> - <strong><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L101">SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L101</a></strong>: ‘subscript(ip_safe:)’ is deprecated: Use <code>ip_safely</code> instead.  <br /><code>
        XCTAssert(testArray[ip_safe: 8] == 9)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/Array+UtilitiesTests.swift#L102">Array+UtilitiesTests.swift#L102</a> - <strong><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L102">SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L102</a></strong>: ‘subscript(ip_safe:)’ is deprecated: Use <code>ip_safely</code> instead.  <br /><code>
        XCTAssertNil(testArray[ip_safe: -1])
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/Array+UtilitiesTests.swift#L103">Array+UtilitiesTests.swift#L103</a> - <strong><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/e0a9fa327349d649162ec408d7591ea17c5357fb/SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L103">SwiftWisdomTests/StandardLibrary/Array/Array+UtilitiesTests.swift#L103</a></strong>: ‘subscript(ip_safe:)’ is deprecated: Use <code>ip_safely</code> instead.  <br /><code>
        XCTAssertNil(testArray[ip_safe: 9])
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 142 tests, with 0 failures (0 unexpected) in 0.241 (0.312) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: SwiftWisdom/Core/Foundation/URL+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//  Created by Max Litteral on 7/17/17.
+//  Copyright © 2017 Intrepid. All rights reserved.
+//
+
+import Foundation
+
+extension URL: ExpressibleByStringLiteral {
+
+    public init(stringLiteral value: StaticString) {
+        self = URL(string: "\(value)")!
```
Comment (vinnyt) [inline SwiftWisdom/Core/Foundation/URL+Extensions.swift:14]: anything we can do to avoid the force unwrap here?

**Review by vinnyt — CHANGES_REQUESTED** [review]

File: SwiftWisdom/Core/Foundation/URL+Extensions.swift (hunk 1 above)
Comment (MaxHasADHD) [reply SwiftWisdom/Core/Foundation/URL+Extensions.swift:14]: I tried looking into that, but there doesn't seem to be a way using `ExpressibleByStringLiteral`. Could create an extension on String to add a `.url` property and have it optional.

Comment (MaxHasADHD) [reply SwiftWisdom/Core/Foundation/URL+Extensions.swift:14]: or, can add https://github.com/JohnSundell/Require pod, which looks like the app will still crash if a bad URL string is used but can attach a useful message.

Comment (alexpersian) [reply SwiftWisdom/Core/Foundation/URL+Extensions.swift:14]: Personally I would rather avoid this addition if it relies on a force unwrap for it's functionality. It's defeating the purpose of Apple's API having the `URL.init` be failable.

**Pull request closed without merging**